### PR TITLE
tastypie OAuth Backend

### PR DESCRIPTION
**Pull request by hishamkaram**
This PR include:
- implement OAuth Backend for tastypie

### CODE REVIEW

File: geonode/api/authentication.py [hunk 1]
```diff
@@ -0,0 +1,59 @@
+# -*- coding: utf-8 -*-
+#########################################################################
+#
+# Copyright (C) 2016 OSGeo
```
Comment (afabiani) [inline geonode/api/authentication.py]: Could you set this to 2019

Comment (hishamkaram) [reply geonode/api/authentication.py]: fixed

**Comment by giohappy** [comment]
it looks good but as @afabiani asked you should include tests

**Comment by hishamkaram** [comment]
@giohappy @afabiani  ok the next two days are holiday in Egypt so it's a good time to write the test cases

**Comment by hishamkaram** [comment]
@afabiani

**Pull request merged**